build.cake

```diff
@@ -0,0 +1,81 @@
+// Arguments.
+var target = Argument("target", "Default");
+var configuration = Argument("configuration", "Release");
+
+// Define directories.
+var solutionFile = GetFiles("./Okta.Xamarin/*.sln").First();
+
+// Android and iOS.
+var androidProject = GetFiles("./Okta.Xamarin/Okta.Xamarin.Android/*.csproj").First();
+var iOSProject = GetFiles("./Okta.Xamarin/Okta.Xamarin.iOS/*.csproj").First();
+
+// Output folders.
+var artifactsDirectory = Directory(System.IO.Path.Combine(Environment.CurrentDirectory, "artifacts"));
+var androidOutputDirectory = Directory(System.IO.Path.Combine(artifactsDirectory, "Android"));
+var iOSOutputDirectory = Directory(System.IO.Path.Combine(artifactsDirectory, "iOS"));
+
+// Tests.
+var testsProject = GetFiles("./Okta.Xamarin/Okta.Xamarin.Test/*.csproj").First();
+
+Task("Clean")
+	.Does(() => 
+	{
+		CleanDirectory(artifactsDirectory);
+
+		DotNetBuild(solutionFile, settings => settings
+			.SetConfiguration(configuration)
+			.WithTarget("Clean")
+			.SetVerbosity(Verbosity.Minimal));
+	});
+
+Task("Restore-Packages")
+	.Does(() => 
+	{
+		NuGetRestore(solutionFile);
+	});
+
+Task("Build-Android")
```

Comment:
Should all build tasks depend on `Restore`? For example:
```
Task("Build-Android")
.IsDependentOn("Restore-Packages")
.Does(() =>
// ...
```

Reply:
I don't understand the question.  Can you clarify?  The build tasks ("Build-Android" and "Build-iOS") each depend on "Restore-Packages".

Reply:
Are you suggesting that they should possibly not require restore?

Reply:
Omg, am I blind? Sorry, I didn't see `IsDependentOn("Restore-Packages")` below.